pkg/apis/core/validation/validation.go

```diff
@@ -5941,6 +5942,57 @@ func validateWindowsSecurityContextOptions(windowsOptions *core.WindowsSecurityC
 	return allErrs
 }
 
+func validateWindowsHostProcessPod(podSpec *core.PodSpec, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	containerCount := 0
+	hostProcessContainerCount := 0
+
+	podHostProcess := false
+	if podSpec.SecurityContext != nil && podSpec.SecurityContext.WindowsOptions != nil && podSpec.SecurityContext.WindowsOptions.HostProcess != nil {
+		podHostProcess = *podSpec.SecurityContext.WindowsOptions.HostProcess
+	}
+
+	hostNetwork := false
+	if podSpec.SecurityContext != nil && podSpec.SecurityContext.HostNetwork == true {
+		hostNetwork = true
+	}
+
+	// Note: Pod-wide WindowsSecurityContextOptions are inherited by containers that do not explicitly set them.
+	for _, c := range append(podSpec.Containers, podSpec.InitContainers...) {
+		containerCount++
+		if c.SecurityContext != nil && c.SecurityContext.WindowsOptions != nil {
+			if c.SecurityContext.WindowsOptions != nil && *c.SecurityContext.WindowsOptions.HostProcess {
+				hostProcessContainerCount++
+			}
+		} else if podHostProcess {
+			hostProcessContainerCount++
+		}
+	}
+
+	if hostProcessContainerCount > 0 {
+		// Fail Pod validation if feature is not enabled instead of dropping fields based on PRR reivew.
+		if !utilfeature.DefaultFeatureGate.Enabled(features.WindowsHostProcessContainers) {
+			errMsg := "Pod contains Windows HostProcess containers but feature gate 'WindowsHostProcessContainers' is not enabled"
+			allErrs = append(allErrs, field.Forbidden(fieldPath, errMsg))
+		}
+
+		// At present Windows Pods must container only HostProcess containers or regular containers.
+		if hostProcessContainerCount != containerCount {
+			errMsg := "If Pod contains any HostProcess containers then all containers must be HostProcess containers"
```

Comment:
Good idea.
Service mesh inject is one of the scenarios we'd like to enable.
Let me see if I can re-work this.

Reply:
I'm finding it a bit difficult to do this cleanly.
If I can't get this done soon I will add a TODO in the code here to fix this before beta.
Would that be acceptable (since we are really close to code-freeze)?

Reply:
I updated the validation to list the names of containers that override HostProcess to be the opposite of what the Pod's HostProcess value is (which is disallowed). PTAL!